### PR TITLE
perf: tree-shake errors and `perf-marks/marks` package

### PR DESCRIPTION
**Pull request by arturovt**
- [x] The commit messages follow these
      [guidelines](https://docs.google.com/document/d/1QrDFcIiPjSLDn3EL15IJygNPiHORgU1_OOAqWjiDU5Y)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** _(check one with "x")_

- [x] Refactoring (no functional changes, no api changes)

**Does this PR introduce a breaking change?** _(check one with "x")_

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by arturovt** [comment]
Hey. Those errors are shown in the console only during development. I'd like them to be tree-shaken away during the production build since they are not needed within the bundle (since they're never shown in production).

I checked the contributing guide, but there's no way to add unit tests for this change. I thought that the PR title and code comments are self-descriptive.

File: projects/ngx-skeleton-loader/src/lib/ngx-skeleton-loader.component.ts [hunk 1]
```diff
@@ -71,8 +77,10 @@ export class NgxSkeletonLoaderComponent implements OnInit, AfterViewInit, OnDest
   }
 
   ngOnInit() {
-    start('NgxSkeletonLoader:Rendered');
-    start('NgxSkeletonLoader:Loaded');
+    if (NG_DEV_MODE) {
```
[the remark this answers is not in the document]
Comment (arturovt) [reply projects/ngx-skeleton-loader/src/lib/ngx-skeleton-loader.component.ts:80]: Could we have a possibility to make it optional? E.g. wrap it with some variable that is true by default (so there's no breaking change), e.g.:
```ts
const PERF_MARKS_ENABLED = typeof perfMarksEnabled === 'undefined' || !!perfMarksEnabled

if (PERF_MARKS_ENABLED) start('NgxSkeletonLoader:Rendered');
```

Those users who don't want that library to be bundled within the production bundle could extend Webpack config with Terser global definitions.

**Comment by willmendesneto** [comment]
@arturovt Could you share why do you need that? As far as I can tell, removing this library or making it optional is not on the plans for this library. However, I'm seeking to understand at this stage since you haven't shared anything relevant about your issue. 

Could you share more details about that, please? Setup, logs, etc?

> I thought that the PR title and code comments are self-descriptive.

Unfortunately, the information in this pull request is quite vague at this stage. That will be helpful since you didn't open an issue here.

**Comment by rhutchison** [comment]
@willmendesneto this PR is to allow the perf-marks/marks lib to be tree shaken during production builds.  The lib will still be a part of the project/package, just not in production.

**Comment by arturovt** [comment]
Hey @willmendesneto 

Let me try to explain in detail. Currently, we show warnings only in development mode since they're guarded with `isDevMode()` condition, but they're still shipped into a bundle when running `ng build --prod`.

`ngDevMode` is a special compile-time variable provided by Angular CLI, Angular itself, Material, and other libraries do tree-shaking in that way, see the following code:

https://github.com/angular/angular/blob/ea61ec25628206d18a424906f685c0d0fd6aa714/packages/platform-browser/src/dom/dom_renderer.ts#L23

Since `ngDevMode` is a compile-time variable, it allows to tree-shake stuff when building for production, which means that `console.*` calls are removed from the production bundle reducing its final size.

This is basically the same as this:

```js
// webpack.config.js
new webpack.DefinePlugin({ THIS_IS_TRUE: JSON.stringify(true) })

// somewhere in code
if (THIS_IS_TRUE === false) {
  // do some stuff here
}
```

As you know, the above `if` branch will be eliminated by Terser when running `webpack --mode production`, since it'll face `if (true === false)` which always defaults to `if (false)`.

The current size of this library (minified) is `6.1kB`; with these changes, it's `4.7kB`.

`perf-marks` will be still shipped with the package; the package will just be tree-shaken away when running `ng build --prod`.

**Comment by markwhitfeld** [comment]
Will this PR be merged to avoid bundling the perf-marks package into a production build?
Is there any reason to include the perf-marks code in production?

**Comment by willmendesneto** [comment]
Hi @markwhitfeld . 

As I mentioned previously, there was no open issues related to the library (as menitoned here https://github.com/willmendesneto/ngx-skeleton-loader/pull/91#issuecomment-954914788) and removing this library or making it optional is not on the plans for this library.

**Pull request closed without merging**